util/settings/settings.go

```diff
@@ -1336,13 +1336,20 @@ func (mgr *SettingsManager) GetSettings() (*ArgoCDSettings, error) {
 		return nil, fmt.Errorf("error parsing Argo CD selector %w", err)
 	}
 	secrets, err := mgr.secrets.Secrets(mgr.namespace).List(selector)
+	var secretsCopy []*apiv1.Secret
+	// SecretNamespaceLister lists all Secrets in the indexer for a given namespace.
+	// Objects returned by the lister must be treated as read-only.
+	// To allow us to modify the secrets, make a copy
+	for _, secret := range secrets {
+		secretsCopy = append(secretsCopy, secret.DeepCopy())
+	}
 	if err != nil {
 		return nil, err
 	}
```

Comment:
Can we move this into a common function to avoid duplicity of code?
Also, I think we should check for errors before making the copy.

Reply:
yes good catch thank you!